packages/@cdktf/hcl2cdk/test/__snapshots__/resource-meta-properties.test.ts.snap

```diff
@@ -0,0 +1,300 @@
+      instanceType: \\"t2.micro\\",
+      lifecycle: {
+        postcondition: [
+          {
+            condition: '\${self.public_dns != \\"\\"}',
+            errorMessage:
+              \\"EC2 instance must be in a VPC that has public DNS hostnames enabled.\\",
+          },
+        ],
+        precondition: [
+          {
+            condition: '\${self.tags[\\"Component\\"] == \\"nomad-server\\"}',
+            errorMessage: \\"Please add the nomad-server tag to this instance.\\",
+          },
+        ],
+      },
+    });
+  }
+}
+"
+`;
+
+exports[`meta-properties provider passing providers typescript snapshot 1`] = `
+"import * as constructs from \\"constructs\\";
+import * as cdktf from \\"cdktf\\";
+/*Provider bindings are generated by running cdktf get.
+See https://cdk.tf/provider-generation for more details.*/
+import * as google from \\"./.gen/providers/google\\";
+class MyConvertedCode extends cdktf.TerraformStack {
+  constructor(scope: constructs.Construct, name: string) {
+    super(scope, name);
+    new google.provider.GoogleProvider(this, \\"google\\", {
+      region: \\"us-central1\\",
+    });
+    const googleEurope = new google.provider.GoogleProvider(this, \\"google_1\\", {
+      alias: \\"europe\\",
+      region: \\"europe-west1\\",
+    });
+    new google.computeInstance.ComputeInstance(this, \\"default\\", {
+      bootDisk: {
+        initializeParams: {
+          image: \\"debian-cloud/debian-11\\",
+          labels: {
+            my_label: \\"value\\",
+          },
+        },
+      },
+      machineType: \\"e2-medium\\",
+      metadata: {
+        foo: \\"bar\\",
+      },
+      metadataStartupScript: \\"echo hi > /test.txt\\",
+      name: \\"test\\",
+      networkInterface: [
+        {
+          accessConfig: [{}],
+          network: \\"default\\",
+        },
+      ],
+      provider: googleEurope,
+      scratchDisk: [
+        {
+          interface: \\"SCSI\\",
+        },
+      ],
+      tags: [\\"foo\\", \\"bar\\"],
+      zone: \\"us-central1-a\\",
+    });
+  }
+}
+"
+`;
+
+exports[`meta-properties provisioner file typescript snapshot 1`] = `
+"import * as constructs from \\"constructs\\";
+import * as cdktf from \\"cdktf\\";
+/*Provider bindings are generated by running cdktf get.
+See https://cdk.tf/provider-generation for more details.*/
+import * as aws from \\"./.gen/providers/aws\\";
+class MyConvertedCode extends cdktf.TerraformStack {
+  constructor(scope: constructs.Construct, name: string) {
+    super(scope, name);
+    new aws.provider.AwsProvider(this, \\"aws\\", {
+      region: \\"us-east-1\\",
+    });
+    new aws.instance.Instance(this, \\"example\\", {
+      ami: \\"ami-a1b2c3d4\\",
+      iamInstanceProfile: \\"aws_iam_instance_profile\\",
+      instanceType: \\"t2.micro\\",
+      provisioners: [
+        {
+          type: \\"file\\",
+          destination: \\"/etc/myapp.conf\\",
+          source: \\"conf/myapp.conf\\",
```

Comment:
Probably out of scope for this PR, but this would need to be converted into an Asset to work, wouldn't it?